packages/lodestar/src/network/metadata/metadata.ts

```diff
@@ -26,22 +29,25 @@ export class MetadataController {
   private enr?: ENR;
   private config: IBeaconConfig;
   private chain: IBeaconChain;
-  private _metadata: phase0.Metadata;
+  private _metadata: {phase0: phase0.Metadata; altair: altair.Metadata};
```

Comment:
You only need to keep a single `altair.Metadata` around since `altair.Metadata` extends `phase0.Metadata`, so it satisfies the `phase0.Metadata` interface.

eg:
```ts
  private _metadata: altair.Metadata;
  ...
  get allPhase0(): phase0.Metadata {
    return this._metadata as phase0.Metadata;
  }
  ...
```

Reply:
If `_metadata.syncnets` gets set though, then when we call `allPhase0` it would still have syncnets in the object even though it was casted to `phase0.Metadata`.  Maybe we just decompose the `_metadata` object and send only the phase0-relevant parts?  like:

```ts
return {attnets: this._metadata.attnets, seqNumber: this._metadata.seqNumber} as phase0.Metadata.
```